### PR TITLE
Fix windows userdata

### PR DESCRIPTION
**Pull request by gabriel-samfira**
This fixes an embarrassing bug of my own making, introduced in commit: 942456eabc885870c5ada502751fad4e589d3a15

where I omitted to remove the corresponding parameters to ```fmt.Sprintf``` after modifying the string it was formatting.

I added a test for ```InstallCommands()``` as well as fixing some tests for password change that seem to have not been caught by the CI. 

Update:

Given that unit agents seem to be gone now, and there is only one service being created on a machine, the machine agent is now run under the local jujud user on windows again.

A followup PR will be created at a later date that will remove code that was used to maintain unit agents on Windows.

## QA steps

```bash
juju bootstrap
juju add-machine --series win2019
```

Wait for machine to boot.

```bash
juju run --machine 0 "sc.exe qc jujud-machine-0"
```

Check that ```SERVICE_START_NAME``` is ```.\jujud```.

### CODE REVIEW

**Comment by hmlanigan** [comment]
check-multi-juju, when running the unit tests on ubuntu, found:
FAIL: userdatacfg_test.go:1387: cloudinitSuite.TestWindowsCloudInit
Here's a snippet of the failure: https://pastebin.ubuntu.com/p/XnTyGfgZ3g/

File: service/windows/service_test.go [hunk 1]
```diff
@@ -212,9 +214,9 @@ func (s *serviceSuite) TestInstallCommands(c *gc.C) {
 	c.Assert(err, gc.IsNil)
 
 	expected := []string{
-		fmt.Sprintf("New-Service -Name %s -DependsOn Winmgmt -DisplayName %s %s", s.name, s.conf.Desc, s.conf.ExecStart),
-		fmt.Sprintf("sc.exe failure %s reset=5 actions=restart/1000", s.name),
-		fmt.Sprintf("sc.exe failureflag %s 1", s.name),
+		fmt.Sprintf("New-Service -Name '%s' -DependsOn Winmgmt -DisplayName '%s' '%s'", s.name, s.conf.Desc, s.conf.ExecStart),
+		fmt.Sprintf("sc.exe failure '%s' reset=5 actions=restart/1000", s.name),
+		fmt.Sprintf("sc.exe failureflag '%s' 1", s.name),
```
Comment (hmlanigan) [inline service/windows/service_test.go]: Use %q instead of '%s'

**Comment by hmlanigan** [comment]
!!build!!

**Comment by hmlanigan** [comment]
@gabriel-samfira please add QA steps to the PR description.

**Comment by gabriel-samfira** [comment]
> @gabriel-samfira,
> 
> During QA with 2.9, the windows machine gets stuck in pending. I'm using OpenStack and the Windows 2012 R2 eval KVM image you pointed me at. The machine is active, but the juju agent never connects to the controller.
> 
> I'm I building the windows agent binaries with:
> $ GOOS=windows make install
> 
> I can successfully add a Windows 2012 R2 eval machine to a 2.8.10 controller on the same OpenStack.

Try the following:

```bash
make release-install
GOOS=windows make release-install
```

Generate simplestreams for agents:

```bash
mkdir -p $HOME/simplestreams/tools/proposed
cp $GOPATH/bin/jujud $HOME/simplestreams/tools/proposed
cp $GOPATH/bin/windows_amd64/jujud.exe $HOME/simplestreams/tools/proposed
cd $HOME/simplestreams/tools/proposed
tar czf juju-2.9.1-ubuntu-amd64.tgz jujud
tar czf juju-2.9.1-windows-amd64.tgz jujud.exe
rm jujud jujud.exe
cd ~
juju-metadata generate-agents -d $HOME/simplestreams --stream proposed
```

Copy the tools folder to a web server that is accessible to your openstack environment, then bootstrap using:

```bash
juju bootstrap --debug --verbose --show-log \
    --config image-metadata-url="http://YOUR_SERVER_IP/images/" \
    --config agent-metadata-url="http://YOUR_SERVER_IP/tools/" \
    --config agent-stream=proposed oss oss-ctrl
```

Once bootstrapped, make sure the checksum of the ```jujud``` binary on the controller matches the one you built (got bit by this one a few times).

If it still fails, please post the log from:

C:\Program Files\Cloudbase Solutions\Cloudbase-Init\log\cloudbase-init.log

Also, on the windows instance, you can check the existance of the windows service representing the agent:

```powershell
Get-Service juju*
```

To get info about the service:

```cmd
sc.exe qc jujud-machine-0
```

**Comment by hmlanigan** [comment]
QA worked today.  Maybe it took the OpenStack I was using to be upgraded?

**Review by hmlanigan — APPROVED** [review]
QA good.  Thank you for the contribution.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**